src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -781,7 +781,7 @@ private static unsafe void Callback(void* configurationContext, void* name, void
 
                 case GCConfigurationType.StringUtf8:
                     {
-                        string? dataAsString = Marshal.PtrToStringUTF8((IntPtr)data);
+                        string? dataAsString = Marshal.PtrToStringUTF8(checked((IntPtr)data));
```

Comment:
Overflow check is not desirable here

Reply:
Worth noting that the overflow check happens for 32-bit code today in .NET 6.

I left a note further down that the GC could likely be updated to just pass in `nint_t` or `void*` and avoid this altogether

Reply:
This is a new code that did not exist in .NET 6 and it was introduced very recently. Adding `checked` here is introducing a bug.